core/src/main/java/org/apache/iceberg/PartitionsTable.java

```diff
@@ -167,16 +175,30 @@ Partition get(StructLike key) {
     private final StructLike key;
     private long recordCount;
     private int fileCount;
+    private int deleteFileCount;
+    private long equalityDeleteCount;
+    private long positionDeleteCount;
 
     Partition(StructLike key) {
       this.key = key;
       this.recordCount = 0;
       this.fileCount = 0;
+      this.deleteFileCount = 0;
+      this.equalityDeleteCount = 0L;
+      this.positionDeleteCount = 0L;
     }
 
-    void update(DataFile file) {
-      this.recordCount += file.recordCount();
+    void update(FileScanTask task) {
+      this.recordCount += task.file().recordCount();
       this.fileCount += 1;
+      this.deleteFileCount += task.deletes().size();
```

Comment:
Would this ever have duplicates?  Example multiple fileScanTask within the same partition referring to the same delete file, can it happen?

Reply:
+1 to this. This could definitely have duplicates!

Reply:
Yes， I'll fix it. Thanks.